# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCClass.h
# Commit: 25adf505426d
# Full Hash: 25adf505426d21f67e15d146f879b0ea53d343a7
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-23 04:09:13
# Regressor Bug: 1642412
# File Overlap Count: 19
# Description:
#   Bug 1642412 - wasm: Rollup of inline table.get/set for anyref. r=lth
#   
#   Rollup of the previous inline table.get/set patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141024
# ==============================================================================

diff -r 43f7556db2e1 -r 25adf505426d js/src/wasm/WasmBCClass.h
--- a/js/src/wasm/WasmBCClass.h	Tue Mar 22 19:44:13 2022 +0000
+++ b/js/src/wasm/WasmBCClass.h	Tue Mar 22 19:59:05 2022 +0000
@@ -1080,6 +1080,15 @@
 
   //////////////////////////////////////////////////////////////////////
   //
+  // Table access.
+
+  Address addressOfTableField(const TableDesc& table, uint32_t fieldOffset,
+                              RegPtr tls);
+  void loadTableLength(const TableDesc& table, RegPtr tls, RegI32 length);
+  void loadTableElements(const TableDesc& table, RegPtr tls, RegPtr elements);
+
+  //////////////////////////////////////////////////////////////////////
+  //
   // Heap access.
 
   void bceCheckLocal(MemoryAccessDesc* access, AccessCheck* check,
@@ -1542,6 +1551,10 @@
   [[nodiscard]] bool emitTableSet();
   [[nodiscard]] bool emitTableSize();
 
+  void emitTableBoundsCheck(const TableDesc& table, RegI32 index, RegPtr tls);
+  [[nodiscard]] bool emitTableGetAnyRef(uint32_t tableIndex);
+  [[nodiscard]] bool emitTableSetAnyRef(uint32_t tableIndex);
+
 #ifdef ENABLE_WASM_GC
   [[nodiscard]] bool emitStructNewWithRtt();
   [[nodiscard]] bool emitStructNewDefaultWithRtt();